src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.test.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import faker from 'faker';
+import times from 'lodash/times';
+
+import { createEnsObject } from 'tests/fixtures/ens-object';
+import {
+  TrackPanelBookmarks,
+  PreviouslyViewedLinks,
+  ExampleLinks
+} from './TrackPanelBookmarks';
+
+import ImageButton from 'src/shared/components/image-button/ImageButton';
+import { PreviouslyViewedObject } from '../../trackPanelState';
+
+jest.mock('react-router-dom', () => ({
+  Link: (props: any) => (
+    <div {...props} className={'link'}>
+      {props.children}
+    </div>
+  )
+}));
+
+const createPreviouslyViewedLink = (): PreviouslyViewedObject => ({
+  genome_id: faker.random.word(),
+  object_id: faker.random.word(),
+  object_type: faker.random.word(),
+  label: faker.random.word()
+});
+
+const closeTrackPanelModalMock = jest.fn();
+const updateTrackStatesAndSaveMock = jest.fn();
+const fetchExampleEnsObjectsMock = jest.fn();
+const changeDrawerViewAndOpenMock = jest.fn();
+
+describe('<TrackPanelBookmarks />', () => {
+  const numberOfExampleObjects = faker.random.number({ min: 5, max: 10 });
+  const numberOfPreviouslyViewedObjects = faker.random.number({
+    min: 5,
+    max: 20
+  });
+
+  const props = {
+    activeGenomeId: faker.random.word(),
+    exampleEnsObjects: times(numberOfExampleObjects, () => createEnsObject()),
+    previouslyViewedObjects: times(numberOfPreviouslyViewedObjects, () =>
+      createPreviouslyViewedLink()
+    ),
+    fetchExampleEnsObjects: fetchExampleEnsObjectsMock,
+    updateTrackStatesAndSave: updateTrackStatesAndSaveMock,
+    closeTrackPanelModal: closeTrackPanelModalMock,
+    changeDrawerViewAndOpen: changeDrawerViewAndOpenMock
+  };
+
+  let wrapper: any;
+
+  beforeEach(() => {
+    wrapper = mount(<TrackPanelBookmarks {...props} />);
+    jest.resetAllMocks();
+  });
+
+  it('renders without any error', () => {
+    expect(wrapper).toHaveLength(1);
+  });
+
+  it('renders correct number of previously viewed links', () => {
+    const previouslyViewedLinksWrapper = wrapper.find(PreviouslyViewedLinks);
+
+    expect(previouslyViewedLinksWrapper.find('.link').length).toBe(
+      numberOfPreviouslyViewedObjects
+    );
+  });
+
+  it('calls closeTrackPanelModal when a previously viewed object link is clicked', () => {
+    const previouslyViewedLinksWrapper = wrapper.find(PreviouslyViewedLinks);
+
+    previouslyViewedLinksWrapper
+      .find('.link')
+      .first()
+      .simulate('click');
+
+    expect(closeTrackPanelModalMock).toBeCalled();
```

Comment:
Can we check the number of times the function is called with `toHaveBeenCalledTimes(1)` rather than whether it is just called?

Reply:
Why do we need to check the number of times it is being called? Isn't that used where we need to check multiple calls?

Reply:
I misunderstood that this was used in a hook. Let's leave like this for now.